pkg/prometheus/agent/statefulset.go

```diff
@@ -361,7 +368,11 @@ func makeStatefulSetService(p *monitoringv1alpha1.PrometheusAgent, config prompk
 func appendAgentArgs(
 	promArgs []monitoringv1.Argument,
 	cg *prompkg.ConfigGenerator,
+	version semver.Version,
 	walCompression *bool) []monitoringv1.Argument {
+	if version.Major == 3 {
```

Comment:
```suggestion
	if cg.version.Major == 3 {
```

Reply:
version was not accessible outside I added a method to do that